src/python/pants/backend/python/target_types.py

```diff
@@ -1401,23 +1401,24 @@ class SetuptoolsSCMWriteToTemplateField(StringField):
     )
 
 
-class SetuptoolsSCMVersion(Target):
-    alias = "setuptools_scm_version"
+class VCSVersion(Target):
+    alias = "vcs_version"
     core_fields = (
         *COMMON_TARGET_FIELDS,
-        SetuptoolsSCMTagRegexField,
-        SetuptoolsSCMDummySourceField,
-        SetuptoolsSCMWriteToField,
-        SetuptoolsSCMWriteToTemplateField,
+        VersionTagRegexField,
+        VCSVersionDummySourceField,
+        VersionGenerateToField,
+        VersionTemplateField,
     )
     help = softwrap(
```

Comment:
I think that this should make explicit that this only works for Python code currently. I could see someone finding it on the website and being confused.

With claiming such a generic name, are we confident that this target will be able to be used for things like JVM?

Reply:
It's pretty generic codegen, the generator happens to be a python tool. Other than that the only thing that is python-centric is the GenerateSourcesRequest. To generate Java we would just need a GenerateSourcesRequest subclass that has `output = JavaSourceField`

Reply:
I don't want to do another CI round just to change a help string, but right now the fact that this is in the experimental python backend is a pretty strong hint...